libs/sdk-ui-ext/src/internal/components/pluggableVisualizations/drillDownUtil.ts

```diff
@@ -0,0 +1,165 @@
+// (C) 2020 GoodData Corporation
+import {
+    IAttribute,
+    IAttributeOrMeasure,
+    IInsight,
+    isAttribute,
+    VisualizationProperties,
+    insightModifyItems,
+    insightReduceItems,
+    attributeLocalId,
+    modifyAttribute,
+    insightSetProperties,
+    insightSetFilters,
+    IFilter,
+    insightProperties,
+    insightItems,
+    bucketItemLocalId,
+    newPositiveAttributeFilter,
+    insightBuckets,
+    bucketsFind,
+    IBucket,
+    bucketIsEmpty,
+    idMatchBucket,
+    uriRef,
+} from "@gooddata/sdk-model";
+import { BucketNames, IDrillEvent } from "@gooddata/sdk-ui";
+import { IImplicitDrillDown } from "../../interfaces/Visualization";
+import { isDrillIntersectionAttributeItem, IDrillEventIntersectionElement } from "@gooddata/sdk-ui";
+import { drillDownFromAttributeLocalId, drillDownDisplayForm } from "../../utils/ImplicitDrillDownHelper";
+import { arrayUtils } from "@gooddata/util";
+
+function matchesDrillDownTargetAttribute(
+    drillDefinition: IImplicitDrillDown,
+    attribute: IAttribute,
+): boolean {
+    return attributeLocalId(attribute) === drillDownFromAttributeLocalId(drillDefinition);
+}
+
+type TColumnWidths = {
+    attributeColumnWidthItem?: {
+        attributeIdentifier: string;
+        width: {
+            value: number;
+        };
+    };
+    measureColumnWidthItem?: {
+        width: {
+            value: number;
+        };
+    };
+};
```

Comment:
pivot table has types for these already, why rewriting here?

Reply:
Also the `T` prefix is strange, we use it for generic parameters.

Reply:
Fixed